dist/tools/kconfiglib/genconfig.py

```diff
@@ -111,57 +122,133 @@ def merge_configs(kconf, configs=[]):
         log(kconf.load_config(config, replace=False))
 
 
-def check_configs(kconf, check_app_sym=False):
+def check_config_symbols(kconf):
+    """
+    Verifies that symbols got the values assigned by the user. This does not
+    check choices. For that, please refer to check_config_choices.
+    """
```

Comment:
Why is this check necessary, when could the symbol not get the value assigned by the user?

Reply:
Because of lacking dependencies, for instance. If an user wants to enable module FOO, which depends on BAR, but BAR is not enabled, then FOO can't be enabled and gets `n`, contrary to what the user set (`y`).

Reply:
Kconfig will not produce an invalid configuration in that case (i.e. FOO without BAR), but we want to tell the user what went wrong...